lib/commons/text/accessible-text.js

```diff
@@ -137,34 +138,34 @@ function nonEmptyText(t) {
  * @return {string}
  */
 text.accessibleText = function(element, inLabelledByContext) {
-	//todo: implement shadowDOM
-	var accessibleNameComputation;
-	var encounteredNodes = [];
+	let accessibleNameComputation;
+	const encounteredNodes = [];
+	if (element instanceof Node) {
```

Comment:
The reason I did this is because it would have taken a whole lot of additional changes to make sure every method that uses accessibleText has access to a virtual node. It wasn't even for backward compatibility, but that's a good additional reason. Further, I think these methods should be useful outside of a rule too, which they aren't currently, since they rely on setup that gets done in another place.

I think, instead of calling `element = axe.utils.getNodeFromTree(axe._tree[0], element);` we need a method that will take any DOM element and return it's virtual counterpart. Something like `virtualNode = axe.utils.getVirtualNode(node)`. It can look at `axe._tree[0]` if available, and if it isn't, it should create a new tree and pull the virtual node from that. We could also make that method store virtual nodes in a set rather than a tree, so that lookup is faster.